lib/improver/wind_gust_diagnostic.py

```diff
@@ -201,18 +201,30 @@ def process(self, cube_gust, cube_ws):
                    ' {0:s} {1:s}.'.format(perc_coord_gust.name(),
                                           perc_coord_ws.name()))
             raise ValueError(msg)
-        # Add metadata to both cubes
+
+        # Check times are compatible.
+        msg = ('Could not match time coordinate')
+        try:
+            wg_time = req_cube_gust.coord('time')
+            ws_time = req_cube_ws.coord('time')
+            for i, point in enumerate(wg_time.points):
```

Comment:
We are making the assumption here that the cubes have the same number of time coordinates and that they are in the same order (gust_time[i] == wind_time[i]). I think this is reasonable as we have tended to standardise the input to IMPROVER, but it would be worth a second opinion.